packages/nouns-webapp/src/index.tsx

```diff
@@ -161,95 +144,39 @@ const Updaters = () => {
   );
 };
 
-const BLOCKS_PER_DAY = 6_500;
-
 const ChainSubscriber: React.FC = () => {
   const dispatch = useAppDispatch();
+  const poolSize = useAppSelector(state => state.auction.config?.poolSize);
 
-  const loadState = async () => {
-    const wsProvider = new WebSocketProvider(config.app.wsRpcUri);
-    const nounsAuctionHouseContract = NounsAuctionHouseFactory.connect(
-      config.addresses.nounsAuctionHouseProxy,
-      wsProvider,
+  // Get the VRGDA auction settings
+  useEffect(() => {
+    getVrgdaAuctionConfig().then(config => {
+      if (config) dispatch(setConfig(config));
+    });
+  }, []);
+
+  // Refresh auction data on new blocks
+  useBlockListener(async blockNumber => {
+    if (typeof poolSize === 'undefined') return;
+
+    const data = await getVrgdaAuctions(blockNumber, poolSize);
```

Comment:
Here querying for current set of auctions on every block. Using JSON provider for it, can potentially optimize it and change to WS if needed.